test/helpers/integration.rb

```diff
@@ -507,30 +538,21 @@ def hot_restart_does_not_drop_connections(num_threads: 1, total_requests: 500)
     msg << "   %4d refused\n"               % replies.fetch(:refused,0)
     msg << "   %4d read timeout\n"          % replies.fetch(:read_timeout,0)
     msg << "   %4d reset\n"                 % replies.fetch(:reset,0)
+    msg << "   %4d write_errors\n"          % replies.fetch(:write_error,0)
     msg << "   %4d success\n"               % replies.fetch(:success,0)
     msg << "   %4d success after restart\n" % replies.fetch(:restart,0)
     msg << "   %4d restart count\n"         % restart_count
 
+    actual_requests = num_threads * num_requests
+    allowed_errors = (actual_requests * 0.002).round
```

Comment:
So 0.2%. 

Looks like Windows allowed 40 errors before and non-Window 1%

We test with between 300 and 3000 requests => we allow 6-60 requests with errors – this is for non-Windows. For Windows we allow any amount of reset and refused requests as long as there's no other errors

Did I get this right?

Reply:
> 300 and 3000 requests => we allow 6-60 requests

Are you off by a factor of ten?  I think it should be:

300 and 3000 requests => we allow 1-6 requests

Re Windows...

First of all, yes, I have used MSFT OS's forever (since DOS days).  But, I don't spend much time testing Puma on Windows. If the tests pass, I'm good.  There have been a lot of changes to test code, so the Windows limits might be able to be lowered.  Not high on my list right now...